src/pip/_internal/req/req_install.py

```diff
@@ -559,11 +559,10 @@ def egg_info_path(self, filename):
                 filenames = [f for f in filenames if f.endswith('.egg-info')]
 
             if not filenames:
-                raise InstallationError(
-                    'No files/directories in %s (from %s)' % (base, filename)
-                )
-            assert filenames, \
-                "No files/directories in %s (from %s)" % (base, filename)
+                error_message="Files/directories (from %s) " + \
+                    "not found in %s " % (filename, base)
+                raise InstallationError(error_message)
+            assert filenames, error_message
```

Comment:
`error_message` is set in the `if not filenames` block, but used outside that block in the `assert` statement. The assert only fires if `filenames` is empty, but the usage is pretty confusing - it looks like `error_message` can be used without being set. Can you fix this please?

Actually, the original code is confusing anyway, as it's not clear how the assert can ever fire, because if it were going to fire, the `InstallationError` would be raised immediately before.

Maybe just remove the assert?

Reply:
Sounds good! (I didn't get how it worked either but didn't plan to update the logic since it seemed to _work_ if not be straightforward. Fix coming.